cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,226 @@
+
+                if (currentQuotas == null) {
+                    // 1. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is not of type `QuotasPluginKafka`, skip alteration
+                    // 2. if the current quotas set in Kafka are null and quotas configuration in Kafka CR is type of `QuotasPluginKafka`, but there is no fields configured, skip alteration
+                    if (isNotKafkaPlugin || ops.stream().allMatch(op -> Objects.isNull(op.value()))) {
+                        LOGGER.debugCr(reconciliation, "There are no default user quotas set in Kafka and the Kafka built-in plugin is not configured, skipping the alteration");
+                        return Future.succeededFuture(false);
+                    } else if (ops.stream().anyMatch(op -> Objects.nonNull(op.value()))) {
+                        // in case that the current quotas are null, but desired quotas contains some non-null values, we should alter the quotas
+                        return Future.succeededFuture(true);
+                    }
+                }
+
+                return Future.succeededFuture(!areCurrentAndDesiredQuotasSame(currentQuotas, ops));
+            });
+    }
+
+    /**
+     * Method for altering the default Kafka user quotas using Kafka Admin client
+     *
+     * @param reconciliation            Reconciliation marker
+     * @param vertx                     Vert.x instance
+     * @param kafkaAdmin                Kafka Admin object
+     * @param clientQuotaAlteration     Quota alteration operation
+     *
+     * @return  Future after completion of the alter operation
+     */
+    private static Future<Void> alterQuotas(
+        Reconciliation reconciliation,
+        Vertx vertx,
+        Admin kafkaAdmin,
+        ClientQuotaAlteration clientQuotaAlteration
+    ) {
+        LOGGER.infoCr(reconciliation, "Altering default user quotas to: {}", clientQuotaAlteration.toString());
+
+        return VertxUtil
+            .kafkaFutureToVertxFuture(reconciliation, vertx, kafkaAdmin.alterClientQuotas(List.of(clientQuotaAlteration)).values().get(DEFAULT_USER_ENTITY))
+            .map((Void) null);
+    }
+
+    /**
+     * Method that compares the desired quotas, represented by List of {@link ClientQuotaAlteration.Op}, and current quotas
+     * set in Kafka, represented by Map.
+     *
+     * @param desiredQuotas     desired quotas, represented as List of {@link ClientQuotaAlteration.Op}
+     * @param currentQuotas     current quotas configured in Kafka, represented as Map
+     *
+     * @return  boolean result of the comparison. Returns true if both current and desired quotas are same, false otherwise
+     */
+    private static boolean areCurrentAndDesiredQuotasSame(Map<String, Double> currentQuotas, List<ClientQuotaAlteration.Op> desiredQuotas) {
+        boolean areSame = true;
+
+        for (ClientQuotaAlteration.Op quota : desiredQuotas) {
```

Comment:
How do you know the `desiredQuota` will always contain all values?

Reply:
The desired quota are built in the method inside this class (`prepareQuotaConfigurationRequest`) and it always has all the values there, only with `null`s in case that user doesn't specify the value of the particular quota.
So it will always has all the values. But to be sure, I can get each of the quotas from the desired and current quotas and compare them. (I originally thought about it, but there was something why I decided to do it this way 🤔 )